src/DebugUrlTracker.php

```diff
@@ -23,15 +23,31 @@ class DebugUrlTracker
 {
     private $url;
 
+    /**
+     * @var Symfony1Context
+     */
+    private $context;
+
+    public function __construct(Symfony1Context $context)
+    {
+        $this->context = $context;
+    }
+
     public function onKernelResponse(FilterResponseEvent $event)
     {
-        if ($this->url || ! $event->isMasterRequest() || ! \sfContext::hasInstance()) {
+        if ($this->url || !$event->isMasterRequest() || !$this->context->isInitialized()) {
             return;
         }
+
         $response_headers = $event->getResponse()->headers;
+
         if ($response_headers->has('x-debug-token-link')
             // Only add when the symfony 1 content type is not javascript
-            && strpos(\sfContext::getInstance()->getResponse()->getContentType(), 'javascript') === false
+            && strpos($this->context->getResponse()->getContentType(), 'javascript') === false
+            // Do not add when response is not HTML or is an attachment
+            && (false !== $response_headers->has('Content-Type') &&
```

Comment:
I think you want to use the headers from the response coming from Symfony 1, instead of 3. The headers are not copied when the Symfony 3 response is created.